people_manager.rb

```diff
@@ -0,0 +1,76 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'base_manager'
+
+class PeopleManager < BaseManager
+  attr_accessor :people
+
+  def initialize
+    super
+    @people = []
+  end
+
+  # List all people
+  def list_all_people
+    check_empty_list(@people, 'people')
+
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
```

Comment:
- [ ] @NicolasAndrehh and @simplegoose great work in refactoring your code, however, I would suggest you move this function to another file, according to the `Single-responsibility Principle` each file should have a single responsibility this will help minimize errors debug quickly and also scale up the app easier in future, in our case all other methods handle inputs but this method does not.

Reply:
Hey @danmainah, we don't get what you mean, that's a method related to people, so it should be inside of peopleManager class, don't you think? Please explain this one a little bit better 🙏